digdag-tests/src/test/java/acceptance/GroupRetryIT.java

```diff
@@ -0,0 +1,583 @@
+                    "-c", config.toString(),
+                    "-e", server.endpoint(),
+                    "--latest-revision",
+                    "--resume-from", "+step2+a",
+                    String.valueOf(originalAttemptId));
+            assertThat(retryStatus.errUtf8(), retryStatus.code(), is(0));
+            retry4 = getAttemptId(retryStatus);
+        }
+
+        // Wait for the attempt to success
+        assertThat(joinAttempt(client, retry4).getSuccess(), is(true));
+
+        assertOutputExists(retry4 + "3-1", false); // skipped
+        assertOutputExists(retry4 + "3-2a", true);
+        assertOutputExists(retry4 + "3-2b", true);
+        assertOutputExists(retry4 + "3-2c", true);
+    }
+
+    @Test
+    public void testGroupRetryWithLoop()
+            throws Exception
+    {
+        DigdagClient client = DigdagClient.builder()
+                .host(server.host())
+                .port(server.port())
+                .build();
+
+        // Push the project
+        pushRevision("acceptance/group_retry/retry-with-loop-1.dig", "test");
+
+        // Start the workflow
+        Id originalAttemptId;
+        {
+            CommandStatus startStatus = main("start",
+                    "-c", config.toString(),
+                    "-e", server.endpoint(),
+                    "group_retry", "test",
+                    "--session", "now");
+            assertThat(startStatus.errUtf8(), startStatus.code(), is(0));
+            originalAttemptId = getAttemptId(startStatus);
+        }
+
+        // Wait for the attempt to fail
+        assertThat(joinAttempt(client, originalAttemptId).getSuccess(), is(false));
+
+        assertOutputExists(originalAttemptId + "1-1", true);
+        assertOutputExists(originalAttemptId + "0-1-2a", true);
+        assertOutputExists(originalAttemptId + "0-1-2b", true);
+        assertOutputExists(originalAttemptId + "0-1-2c", false);
+
+        // Push the project
+        pushRevision("acceptance/group_retry/retry-with-loop-2.dig", "test");
+
+        // Retry without updating the revision: --keep-revision
+        Id retry1;
+        {
+            CommandStatus retryStatus = main("retry",
+                    "-c", config.toString(),
+                    "-e", server.endpoint(),
+                    "--keep-revision",
+                    "--all",
+                    String.valueOf(originalAttemptId));
+            assertThat(retryStatus.errUtf8(), retryStatus.code(), is(0));
+            retry1 = getAttemptId(retryStatus);
+        }
+
+        // Wait for the attempt to fail again
+        assertThat(joinAttempt(client, retry1).getSuccess(), is(false));
+
+        assertOutputExists(retry1 + "1-1", true);
+        assertOutputExists(retry1 + "0-1-2a", true);
+        assertOutputExists(retry1 + "0-1-2b", true);
+        assertOutputExists(retry1 + "0-1-2c", false);
+
+        // Retry with the latest fixed revision & resume failed
```

Comment:
resume failed ?

Reply:
resume failed actually means retry failed tasks. My understanding that it comes from here: https://github.com/treasure-data/digdag/blob/1f277a06b9ada4c9bbf1c6884f9e3ff20ca1af52/digdag-server/src/main/java/io/digdag/server/rs/AttemptResource.java#L220-L224

Reply:
So `resume failed tasks` or something is clearer?